File: Lab5/5.1/c_sharp/Segment.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace OOP_lab5
{
    public class Segment : Line
    {
        public Segment(Coord start, Coord end): base(start,end){ }

        public void CuttingSegment()
        {
            end.x = end.x - 5*((end.x-start.x)/Length());
            end.y = end.y - 5 * ((end.y - start.y) / Length());
        }

        public Coord GetStartCoords()=> start;
        public Coord GetEndCoords() => end;
    }
}

File: Lab5/5.1/c++/Segment.cpp
#include "Segment.h"

Segment::Segment(Coord start, Coord end) : Line(start, end) { }

void Segment::CuttingSegment()
{
    end.x = end.x - 5 * ((end.x - start.x) / Length());
    end.y = end.y - 5 * ((end.y - start.y) / Length());
}
Coord Segment::GetStartCoords() 
{ return start; }
Coord Segment::GetEndCoords() 
{ return end; }